File: features/landing/src/main/java/com/walhalla/landing/config/Cfg.java
package com.walhalla.landing.config;

public class Cfg {

    private final boolean toolbarEnabled;
    private final long counterTimeMs;
    private final boolean splashScreenEnabled;

    public void setCheckConnection(boolean checkConnection) {
        this.checkConnection = checkConnection;
    }

    private boolean checkConnection;
    private final boolean swipeEnabled;
    private final boolean progressEnabled;

    private Cfg(Builder builder) {
        this.toolbarEnabled = builder.toolbarEnabled;
        this.counterTimeMs = builder.counterTimeMs;
        this.splashScreenEnabled = builder.splashScreenEnabled;
        this.checkConnection = builder.checkConnection;
        this.swipeEnabled = builder.swipeEnabled;
        this.progressEnabled = builder.progressEnabled;
    }

    public boolean isToolbarEnabled() {
        return toolbarEnabled;
    }

    public long getCounterTimeMs() {
        return counterTimeMs;
    }

    public boolean isSplashScreenEnabled() {
        return splashScreenEnabled;
    }

    public boolean isCheckConnection() {
        return checkConnection;
    }

    public boolean isSwipeEnabled() {
        return swipeEnabled;
    }

    public boolean isProgressEnabled() {
        return progressEnabled;
    }

    public static class Builder {

        private boolean toolbarEnabled;
        private long counterTimeMs;
        private boolean splashScreenEnabled;
        private boolean checkConnection;
        private boolean swipeEnabled;
        private boolean progressEnabled;

        public Builder() {
            // Установка значений по умолчанию
            this.toolbarEnabled = false;
            this.counterTimeMs = 2400;
            this.splashScreenEnabled = true;
            this.checkConnection = true;
            this.swipeEnabled = false;
            this.progressEnabled = true;
        }

        public Builder setToolbarEnabled(boolean toolbarEnabled) {
            this.toolbarEnabled = toolbarEnabled;
            return this;
        }

        public Builder setCounterTimeMs(long counterTimeMs) {
            this.counterTimeMs = counterTimeMs;
            return this;
        }

        public Builder setSplashScreenEnabled(boolean splashScreenEnabled) {
            this.splashScreenEnabled = splashScreenEnabled;
            return this;
        }

        public Builder setCheckConnection(boolean checkConnection) {
            this.checkConnection = checkConnection;
            return this;
        }

        public Builder setSwipeEnabled(boolean swipeEnabled) {
            this.swipeEnabled = swipeEnabled;
            return this;
        }

        public Builder setProgressEnabled(boolean progressEnabled) {
            this.progressEnabled = progressEnabled;
            return this;
        }

        public Cfg build() {
            return new Cfg(this);
        }
    }
}

File: features/landing/src/main/java/com/walhalla/landing/config/Cfg.kt
package com.walhalla.landing.config

class Cfg private constructor(builder: Builder) {
    val isToolbarEnabled: Boolean
    val counterTimeMs: Long
    val isSplashScreenEnabled: Boolean

    var isCheckConnection: Boolean
    val isSwipeEnabled: Boolean
    val isProgressEnabled: Boolean

    init {
        this.isToolbarEnabled = builder.toolbarEnabled
        this.counterTimeMs = builder.counterTimeMs
        this.isSplashScreenEnabled = builder.splashScreenEnabled
        this.isCheckConnection = builder.checkConnection
        this.isSwipeEnabled = builder.swipeEnabled
        this.isProgressEnabled = builder.progressEnabled
    }

    // Установка значений по умолчанию
    class Builder {
        var toolbarEnabled = false
        var counterTimeMs: Long = 2400
        var splashScreenEnabled = true
        var checkConnection = true
        var swipeEnabled = false
        var progressEnabled = true

        fun setToolbarEnabled(toolbarEnabled: Boolean): Builder {
            this.toolbarEnabled = toolbarEnabled
            return this
        }

        fun setCounterTimeMs(counterTimeMs: Long): Builder {
            this.counterTimeMs = counterTimeMs
            return this
        }

        fun setSplashScreenEnabled(splashScreenEnabled: Boolean): Builder {
            this.splashScreenEnabled = splashScreenEnabled
            return this
        }

        fun setCheckConnection(checkConnection: Boolean): Builder {
            this.checkConnection = checkConnection
            return this
        }

        fun setSwipeEnabled(swipeEnabled: Boolean): Builder {
            this.swipeEnabled = swipeEnabled
            return this
        }

        fun setProgressEnabled(progressEnabled: Boolean): Builder {
            this.progressEnabled = progressEnabled
            return this
        }

        fun build(): Cfg {
            return Cfg(this)
        }
    }
}

